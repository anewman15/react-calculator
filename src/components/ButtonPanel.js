/*
 eslint-disable no-unused-vars
*/

import React from 'react';
import Button from './Button';

const ButtonPanel = props => {
  const btnGroups = [
    ['AC', '+/-', '%', '/'],
    ['7', '8', '9', 'X'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '='],
  ];

  const handleClick = props.clickHandler;

  const keyButtons = btnGroups.map(group => (
    <div key={group.toString()}>
      {group.map(name => (
        <Button key={name.toString()} btnName={name} clickHandler={handleClick}/>
      ))}
    </div>
  ));

  return (
    <div>
      {keyButtons}
    </div>
  );
};

export default ButtonPanel;