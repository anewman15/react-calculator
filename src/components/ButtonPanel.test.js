/*
 eslint-disable no-unused-vars, import/named, import/no-cycle
*/

import React from 'react';
import { act, render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import ButtonPanel from './Button';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('ButtonPanel', () => {
  it('should render groups of buttons', () => {
    act(() => {
      render(<ButtonPanel />, container);

      expect(pretty(container.innerHTML))
        .toMatchSnapshot();
    });
  });
});
