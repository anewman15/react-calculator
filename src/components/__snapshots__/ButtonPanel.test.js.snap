// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel should render groups of buttons 1`] = `""`;
